src/fides/api/task/graph_task.py

```diff
@@ -815,17 +815,18 @@ def build_consent_dataset_graph(datasets: List[DatasetConfig]) -> DatasetGraph:
     consent_datasets: List[GraphDataset] = []
 
     for dataset_config in datasets:
-        connection_type: ConnectionType = (
-            dataset_config.connection_config.connection_type  # type: ignore
-        )
-        saas_config: Optional[Dict] = dataset_config.connection_config.saas_config
-        if (
-            connection_type == ConnectionType.saas
-            and saas_config
-            and saas_config.get("consent_requests")
-        ):
+        connection_config: ConnectionConfig = dataset_config.connection_config
+
+        if connection_config.connection_type != ConnectionType.saas:
+            continue
+
+        saas_config = connection_config.get_saas_config()
+        if not saas_config:
+            continue
+
+        if ActionType.consent in saas_config.supported_actions:
```

Comment:
Thanks for updating this @galvana ! From what I understand, our check that `saas_config.get("consent_requests")` exists was too restrictive, because for bi-directional / notice-based consent, we won't have any consent requests defined in the yml, and instead we rely on the `supported_actions` which is based on whether or not we have that `CONSENT_UPDATE` override fn defined. Is that correct?